File: longest-palindromic-substring/longest-palindromic-substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        int n= s.size();
        if(n==0){
            return "";
        }
        bool dp[n][n]; 
        memset(dp,0,sizeof(dp)); 
        for(int i=0;i<n;i++){
            dp[i][i]=true;
        }
        string str ="";
        str+=s[0];
        for(int i=n-1;i>=0;i--){
            for(int j=i+1;j<n;j++){
                if(s[i]==s[j]){
                    if(j-i == 1 || dp[i+1][j-1]){
                        dp[i][j] = true;
                        if(str.size() < (j - i + 1)){
                            str = s.substr(i,j-i+1);
                        }
                    }
                }
            }
        }
        return str;
    }
};

File: longest-palindromic-substring/longest-palindromic-substring.java
class Solution {
    public String longestPalindrome(String s) {
        int len = 0;
        String str ="";
        for(int i =0; i < s.length(); i++){
            for(int j = i; j <= s.length(); j++){
                if(isPalindrome(s.substring(i,j)) && j-i > len){
                    len = j-i;
                    str = s.substring(i,j);
                }
            }
        }
        return str;
    }
    public boolean isPalindrome(String str){
        int i = 0; 
        int j = str.length() -1;
        while(i<=j){
            if(str.charAt(i++) != str.charAt(j--)){
                return false;
            }
        }
        return true;
    }
}